crates/bevy_ecs/src/storage/sparse_set.rs

```diff
@@ -137,82 +139,84 @@ impl ComponentSparseSet {
     /// The `value` pointer must point to a valid address that matches the [`Layout`](std::alloc::Layout)
     /// inside the [`ComponentInfo`] given when constructing this sparse set.
     pub unsafe fn insert(&mut self, entity: Entity, value: OwningPtr<'_>, change_tick: u32) {
-        if let Some(&dense_index) = self.sparse.get(entity) {
-            self.dense.replace_unchecked(dense_index, value);
-            *self.ticks.get_unchecked_mut(dense_index) =
+        if let Some(&dense_index) = self.sparse.get(entity.id()) {
+            self.dense.replace_unchecked(dense_index as usize, value);
```

Comment:
Is there a cost incurred in this cast in Rust?

Reply:
Casts in rust are primitive enough that their effects on performance heavily depend on what the compiler does to them. Probably worth benchmarking this PR, but i can't imagine this cast would be significant in any way.

Reply:
Added the benchmarks below.